HMX: add disable wake up source for in_start and digital in (#52)
hostmobility/documentation: docs/interfaces/hmx/digital_io.md

```
echo gpio-keys > /sys/bus/platform/drivers/gpio-keys/unbind
```
After waking up, you can rebind if you want to use the gpio-keys.

You can disable one or more wake-up sources using signal-specific macro codes.

| Hardware Name           | Software Name | Key Code | Macro Code |
|-------------------------|---------------|----------|------------|
| Digital_input_start     | in_start      | 656      | KEY_MACRO1 |
| Digital_input_pulldown_1| in_pulldown1  | 657      | KEY_MACRO2 |
| Digital_input_pulldown_2| in_pulldown2  | 658      | KEY_MACRO3 |
| Digital_input_pullup_1  | in_pullup1    | 659      | KEY_MACRO4 |
| Digital_input_pullup_2  | in_pullup2    | 660      | KEY_MACRO5 |
| Digital_input_HMI_1     | in_hmi1       | 661      | KEY_MACRO6 |
| Digital_input_HMI_2     | in_hmi2       | 662      | KEY_MACRO7 |

Example on how to disable in_start (all start signals):
```
echo 656 | tee /sys/bus/platform/drivers/gpio-keys/gpio-keys/disabled_keys
```
It is also possible, within limits, to see what caused the wake up.

> The following example displays all wakeup_count instances where gpio-keys is wakeup4 (wakeup4 has all digital in together so you can not see which signal that has been triggerd).

Reviewer: ```suggestion
The following example displays all wakeup_count instances where gpio-keys is wakeup4 (wakeup4 has all digital in together so here you cannot see which signal was triggered).
```

## Sleep and wake up

> For Digital in, wake-up is default on. If you want to disable the inputs one way is to change the device tree when building the image, the other is to unbind the gpio-keys.
> Disable all input wake ups.

Reviewer: ```suggestion
For Digital in, wake-up is enabled by default. If you want to disable this feature, one way is to modify the device tree when building the image, the other is to unbind the gpio-keys as follows.
```

```
echo gpio-keys > /sys/bus/platform/drivers/gpio-keys/unbind
```

> After waking up, you can bind it again if you want to use the gpio-keys.

Reviewer: ```suggestion
After waking up, you can rebind if you want to use the gpio-keys.
```

> Disable one or more wake up sources using using the macro code number for that signal

Reviewer: ```suggestion
You can disable one or more wake-up sources using signal-specific macro codes.
```

| Hardware Name           | Software Name | Key Code | Macro Code |
|-------------------------|---------------|----------|------------|

> | Digital_input_start     | in_start      | 656      | KEY_MACRO1	|

Reviewer: ```suggestion
| Digital_input_start     | in_start      | 656      | KEY_MACRO1 |
```

| Digital_input_pulldown_1| in_pulldown1  | 657      | KEY_MACRO2 |
| Digital_input_pulldown_2| in_pulldown2  | 658      | KEY_MACRO3 |
| Digital_input_pullup_1  | in_pullup1    | 659      | KEY_MACRO4 |
| Digital_input_pullup_2  | in_pullup2    | 660      | KEY_MACRO5 |
| Digital_input_HMI_1     | in_hmi1       | 661      | KEY_MACRO6 |
| Digital_input_HMI_2     | in_hmi2       | 662      | KEY_MACRO7 |

> Example disable in_start (all start signals)

Reviewer: ```suggestion
Example on how to disable in_start (all start signals):
```

```
echo 656 | tee /sys/bus/platform/drivers/gpio-keys/gpio-keys/disabled_keys
```

> It is also possible within limit to see what cause the wake up.
> All wakeup_count (gpio-keys is wakeup4)

Reviewer: ```suggestion
It is also possible, within limits, to see what caused the wake up.
The following displays all wakeup_count instances (gpio-keys is wakeup4)
```

Reviewer: I'm not entirely sure what is meant by limit here.

Author: ```suggestion
It is also possible, within limits, to see what caused the wake up.
The following example displays all wakeup_count instances where gpio-keys is wakeup4 (wakeup4 has all digital in together so you can not see which signal that has been triggerd).
```
